### PR TITLE
[charts/gateway] US954087_otel_redis

### PR DESCRIPTION
**Pull request by Gazza7205**
**Description of the change**
## 3.0.27 General Updates
- Default image updated to v11.1.00
- Added preview support for [OpenTelemetry](https://opentelemetry.io/)
  - Please see [Techdocs](https://techdocs.broadcom.com/us/en/ca-enterprise-software/layer7-api-management/api-gateway/11-1/install-configure-upgrade/configuring-opentelemetry-for-the-gateway.html) for more details about this integration
  - Preview feature (only available on Gateway v11.1.00)
  - Created an integration example [here](https://github.com/Layer7-Community/Integrations/tree/main/grafana-stack-prometheus-otel) that details how to deploy and configure observability backend to use with the Gateway
    - OpenTelemetry is supported by [numerous vendors](https://opentelemetry.io/ecosystem/vendors/)
      - You are ***not required*** to use the observability stack that we provide as an example.
      - The observability stack that we provide ***is not*** production ready and should be used solely as an example or reference point.
  - Gateway Chart configuration [options](#open-telemetry)
- Redis standalone now supports TLS and Password auth (only available on Gateway v11.1.00)
  - see [Redis configuration](#redis-configuration)

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[charts/gateway]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: charts/gateway/values.yaml
Comment (burbanski) [inline charts/gateway/values.yaml:1]: Please verify recommended changes with @alisterz & @jchanbcbc . Also make final recommended changes to production-values.yaml too.

First, verify that with probable method of injecting the agent into the container gateway the following are not needed under config.javaArgs like they are needed for other form factors (and commented out with a comment that they should only be uncommented if deploying the agent; and perhaps also updated for the specific agent jar version):

- -javaagent:/opt/SecureSpan/Gateway/runtime/lib/opentelemetry-javaagent-2.xx.x.jar
- -Dotel.javaagent.configuration-file=/opt/SecureSpan/Gateway/node/default/etc/conf/system.properties

Second, from a config.systemProperties perspective, we should should have two sets of system properties, one for minimal required for using OTel with only the SDK and another with additional properties if optionally deploying the agent. There should be comments at the top of each section to make clear which sections a customer should uncomment depending on their circumstances. Right now, we're mixing the two.

**Review by burbanski — APPROVED** [review]
Changes to the readme and values files look good to me.

**Review by alisterz — APPROVED** [review]

File: charts/gateway/README.md [hunk 1]
```diff
@@ -61,11 +61,58 @@ Helm Version    Supported Kubernetes Versions
 * [Upgrade the Chart](#upgrading-the-chart)
```
Comment (jchanbcbc) [inline charts/gateway/README.md:61]: For L2, do we need to add Redis?  Is that optional?

Comment (jchanbcbc) [reply charts/gateway/README.md:61]: Also for L2 does mentioning mysql contradict the Important Note below?

File: charts/gateway/README.md [hunk 2]
```diff
@@ -61,11 +61,58 @@ Helm Version    Supported Kubernetes Versions
...
+* [RBAC Parameters](#rbac-parameters)
+* [Service Metrics Demo](#service-metrics-demo)
+* [SubChart Configuration](#subchart-configuration)
+
+# Java 17
+The Layer7 API Gateway is now running with Java 17 with the release of v11.1.00.
+
+If you use Policy Manager, you will need to update to v11.1.00.
+
 # Java 11
```
Comment (jchanbcbc) [inline charts/gateway/README.md:93]: Do we still need to reference GW 10.1? 10.1 will be EOL soon.  Maybe mention 11.0 instead?

File: charts/gateway/README.md [hunk 3]
```diff
@@ -838,6 +862,67 @@ ingress:
...
 | `pmtagger.containerSecurityContext`    | [Container Security Context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container)          | `{}` |
 
+### OpenTelemetry Configuration
+The Gateway from v11.1.00 can be configured to send telemetry to Observability backends [that support OpenTelemetry](https://opentelemetry.io/ecosystem/vendors/). Please see [Techdocs](https://techdocs.broadcom.com/us/en/ca-enterprise-software/layer7-api-management/api-gateway/11-1/install-configure-upgrade/configuring-opentelemetry-for-the-gateway.html) for more details about this integration.
+
+This feature is a ***preview feature*** for v11.1.00 and is ***intentionally disabled*** by default. As with any integration that generates telemetry, there is a performance drop when turning on the OpenTelemetry integration with all of the features enabled.
+
+There is an integration example avaialable [here](https://github.com/Layer7-Community/Integrations/tree/main/grafana-stack-prometheus-otel) that details how to deploy and configure an observability backend to use with the Gateway.
```
Comment (jchanbcbc) [inline charts/gateway/README.md]: typo

File: charts/gateway/README.md [hunk 4]
```diff
@@ -838,6 +862,67 @@ ingress:
...
+otel.logs.exporter=none
+```
+- cluster-wide properties
+```
+otel.enabled=true
+otel.serviceMetricEnabled=true
+otel.traceEnabled=true (if tracing is required)
+otel.traceConfig=(default {})
+```
+example otel.traceConfig
```
Comment (jchanbcbc) [inline charts/gateway/README.md:901]: For this section, may want to reference Techdocs to explain better the possible configuration values? Esp the excluding of assertions as it requires entering the assertion name (with the correct spacing) and where to find the name from Policy Manager.

**Review by jchanbcbc — APPROVED** [review]
Please see my comments. Otherwise looks good.

File: charts/gateway/README.md (hunk 2 above)
Comment (Gazza7205) [reply charts/gateway/README.md:93]: Good point, I'll save that for a PR nearer EOL.

File: charts/gateway/README.md (hunk 4 above)
Comment (Gazza7205) [reply charts/gateway/README.md:901]: Link added under the section heading

File: charts/gateway/README.md (hunk 1 above)
Comment (Gazza7205) [reply charts/gateway/README.md:61]: Added Redis, it is optional

File: charts/gateway/README.md (hunk 3 above)
Comment (Gazza7205) [reply charts/gateway/README.md]: updated

**Pull request merged**